NUEntity.js

```diff
@@ -107,12 +107,11 @@ export default class NUEntity extends NUObject {
         const attributeDescriptors = this.constructor.attributeDescriptors;
         Object.entries(attributeDescriptors).forEach(([localName, attributeObj]) => {
             if (attributeObj.remoteName in JSONObject) {
+                const value = JSONObject[attributeObj.remoteName];
                 if (attributeObj.attributeType == NUAttribute.ATTR_TYPE_INTEGER || attributeObj.attributeType == NUAttribute.ATTR_TYPE_FLOAT) {
-                    this[localName] = Number(JSONObject[attributeObj.remoteName]);
-                }
-                else
-                {
-                    this[localName] = JSONObject[attributeObj.remoteName];
+                    this[localName] = (!value && value !== 0 && attributeObj.subType) ? null : Number(value);
```

Comment:
you don't need value !== 0 !value is enough

Reply:
if the attribute value is 0, !value would return true, resulting in `null` being set. hence added the !==0 check